pkg/sbom/cyclonedx/unmarshal.go

```diff
@@ -103,6 +109,39 @@ func (b *BOM) parseMetadataComponent(bom *cdx.BOM) (*core.Component, error) {
 	return root, nil
 }
 
+func (b *BOM) parseExternalReferences(bom *cdx.BOM) []core.ExternalReference {
+	if bom.ExternalReferences == nil {
+		return nil
+	}
+	var refs = make([]core.ExternalReference, 0)
+
+	for _, ref := range *bom.ExternalReferences {
+		t, err := b.unmarshalReferenceType(ref.Type)
+		if err != nil {
+			continue
+		}
+
+		externalReference := core.ExternalReference{
+			Type: t,
+			URL:  ref.URL,
+		}
+
+		refs = append(refs, externalReference)
+	}
+	return refs
+}
+
+func (b *BOM) unmarshalReferenceType(t cdx.ExternalReferenceType) (core.ExternalReferenceType, error) {
+	var referenceType core.ExternalReferenceType
+	switch t {
+	case cdx.ERTypeExploitabilityStatement:
+		referenceType = core.ExternalReferenceVEX
+	default:
+		return "", ErrUnsupportedExternalReference
```

Comment:
I don't think we need to define `ErrUnsupportedExternalReference`.Since `ErrUnsupportedType` is handled by the upstream caller, other types of errors may also be returned, so `ErrUnsupportedType` was necessary to ensure accurate identification of the error type. However, in this case, that is not necessary, so I think it is sufficient to simply return an error and continue if it is not nil. It can be defined at that time if it becomes necessary in the future.